### PR TITLE
test: add ability to control experimental tests

### PR DESCRIPTION
**Pull request by mhdawson**
Add the ability to specify a NAPI_VERSION which
limits the tests executed to those supported by that
version.

As an example tests can be built/run as:
  npm test --NAPI_VERSION=3

Fixes: https://github.com/nodejs/node-addon-api/issues/349

### CODE REVIEW

**Comment by mhdawson** [comment]
FYI, passed CI runs on my branch for 10.x, 8.x. 6.x and master.

File: napi-inl.h [hunk 1]
```diff
@@ -298,7 +298,9 @@ inline bool Value::IsNumber() const {
   return Type() == napi_number;
 }
 
-#ifdef NAPI_EXPERIMENTAL
+// currently experimental guard with version of NAPI_VERSION that it is 
+// released in once it is no longer experimental
+#if (NAPI_VERSION > 2147483646)
```
Comment (devsnek) [inline napi-inl.h]: isn't `NAPI_EXPERIMENAL` already defined as `NAPI_VERSION > 2147483646`

File: test/index.js [hunk 2]
```diff
@@ -31,11 +31,24 @@ let testModules = [
...
   'objectreference',
   'version_management'
 ];
 
+if((process.env.npm_config_NAPI_VERSION !== undefined) &&
+   (process.env.npm_config_NAPI_VERSION < 50000)) {
+  // currently experimental only test if NAPI_VERSION
+  // is set to experimental. We can't use C max int 
+  // as that is not supported as a number on earlier
+  // Node.js versions.  Oonce bigint is in a release
```
Comment (thefourtheye) [inline test/index.js]: Nit: Once

File: napi-inl.h (hunk 1 above)
Comment (thefourtheye) [reply napi-inl.h]: I am trying to understand this correctly. I also had a similar question in mind when I saw this patch. Why can't we conditionally define `NAPI_EXPERIMENTAL`?

File: test/index.js [hunk 3]
```diff
@@ -31,11 +31,24 @@ let testModules = [
   'object/set_property',
   'promise',
   'typedarray',
+  'typedarray-bigint',
   'objectwrap',
   'objectreference',
   'version_management'
 ];
 
+if((process.env.npm_config_NAPI_VERSION !== undefined) &&
```
Comment (romandev) [inline test/index.js]: nit: if (
(one space)

Comment (mhdawson) [reply test/index.js]: Thanks for catching that.

File: test/index.js (hunk 2 above)
Comment (mhdawson) [reply test/index.js]: good catch

**Comment by mhdawson** [comment]
Landed as https://github.com/nodejs/node-addon-api/commit/73fed84ceb51dea3993179885009e86c9630eb12

**Comment by mhdawson** [comment]
CI run that tests across Node.js versions: https://ci.nodejs.org/view/x%20-%20Abi%20stable%20module%20API/job/node-test-node-addon-api-LTS%20versions/80/

**Comment by mhdawson** [comment]
CI back to green ,closing.

**Pull request closed without merging**